### ISSUE TITLE
How to link to install-name rpath version of libraries for rootless?

### ISSUE THREAD

**Issue by r-plus** [theos/theos#716]
### What are the steps to reproduce this issue?

1. Create the tweak
2. add library link configuration in Makefile like this

```
THEOS_PACKAGE_SCHEME = rootless

TWEAK_NAME = DoubleCut
DoubleCut_FILES = Tweak.xm
DoubleCut_LDFLAGS = -L$(THEOS_VENDOR_LIBRARY_PATH)/$(THEOS_TARGET_NAME)/$(or $(THEOS_PACKAGE_SCHEME),rootful) -lprefs
```

in this case, `libprefs.tbd` is located in both of `vendor/lib/libprefs.tbd` and `vendor/lib/iphone/rootless/libprefs.tbd`

3. make then check the linking library by `otool`

### What happens?

linked libprefs.dylib is not rootless (rpath) version.

```
$ otool -L .theos/obj/debug/arm64e/DoubleCut.dylib
.theos/obj/debug/arm64e/DoubleCut.dylib:
	/Library/MobileSubstrate/DynamicLibraries/DoubleCut.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	/System/Library/Frameworks/Foundation.framework/Foundation (compatibility version 300.0.0, current version 1774.101.0)
	/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation (compatibility version 150.0.0, current version 1774.101.0)
	@rpath/CydiaSubstrate.framework/CydiaSubstrate (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 904.4.0)
	/usr/lib/libprefs.dylib (compatibility version 2.2.0, current version 2.2.3)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
```

### What were you expecting to happen?

```
	/usr/lib/libprefs.dylib (compatibility version 2.2.0, current version 2.2.3)
```

should be

```
	@rpath/libprefs.dylib (compatibility version 2.2.0, current version 2.2.3)
```

### Paste any relevant logs, error output, etc.

```shell
When I temporary rename the `vendor/lib/libprefs.tbd`, result is ok.


$ mv $THEOS/vendor/lib/libprefs.tbd{,.bk}
$ make
omit build logs...
$ otool -L .theos/obj/debug/arm64e/DoubleCut.dylib
.theos/obj/debug/arm64e/DoubleCut.dylib:
	/Library/MobileSubstrate/DynamicLibraries/DoubleCut.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	/System/Library/Frameworks/Foundation.framework/Foundation (compatibility version 300.0.0, current version 1774.101.0)
	/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation (compatibility version 150.0.0, current version 1774.101.0)
	@rpath/CydiaSubstrate.framework/CydiaSubstrate (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 904.4.0)
	@rpath/libprefs.dylib (compatibility version 2.2.0, current version 2.2.3)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
```

I'm thinking this is in development issue to move the libraries from `vendor/lib/<all libraries for rootful>` to `vendor/lib/iphone/rootful`, right?

### What OS are you seeing the problem on?

macOS

### What OS version does it have installed?

macOS 13.3

### What toolchain and version are you using?

Xcode 14.3

### Which SDK version are you using?

iOS 14.4

### What OS is your client device running?

iOS

### What OS version does it have installed?

n/a

### Which device model is it?

n/a

**Comment by L1ghtmann** [comment]
Thanks for catching this! Does it happen to be resolved by #718?

**Comment by r-plus** [comment]
ya, resolved.
